userguide/platform/userguide_configuration.adoc

```diff
@@ -62,178 +95,296 @@ curl -v \
+
+(Replace `<instancename>` with the notification queue instance name (typically, `main` as specified <<notification_queue,above>>)  |Number of threads to use   |10   
+|Notification Queue   |org.killbill.notificationq.${instanceName}.claimed + 
+
+(Replace `<instancename>` with the notification queue instance name (typically, `main` as specified <<notification_queue,above>>)  |Number of notifications to fetch at once   |10   
+|Persistent Bus   |org.killbill.persistent.bus.${instanceName}.sleep + 
+
+(Replace `<instancename>` with the persistent bus instance name (typically, `main` or `external` as specified <<persistent_bus,above>>)  |Time in milliseconds to sleep between runs (only valid in STICKY_POLLING, POLLING).   |3000   
+|Persistent Bus   |org.killbill.persistent.bus.${instanceName}.queue.mode + 
+
+(Replace `<instancename>` with the persistent bus instance name (typically, `main` or `external` as specified <<persistent_bus,above>>)  |How entries are put in the queue. Possible values are  `STICKY_EVENTS`, `STICKY_POLLING` ,`POLLING` as specified in https://github.com/killbill/killbill-commons/blob/4b8d8e647b8028076b5e74cdc27170bda50ca029/queue/src/main/java/org/killbill/queue/api/PersistentQueueConfig.java[PersistentQueueMode]    |STICKY_EVENTS   
+|Persistent Bus   |org.killbill.persistent.bus.${instanceName}.nbThreads  + 
+
+(Replace `<instancename>` with the persistent bus instance name (typically, `main` or `external` as specified <<persistent_bus,above>>)  |Max number of dispatch threads to use |30   
+|Persistent Bus   |org.killbill.persistent.bus.${instanceName}.claimed + 
+
+(Replace `<instancename>` with the persistent bus instance name (typically, `main` or `external` as specified <<persistent_bus,above>>)   |Number of bus events to fetch from the database at once (only valid in 'polling mode')   |10   
+|Persistent Bus   |org.killbill.persistent.bus.${instanceName}.queue.capacity + 
+
+(Replace `<instancename>` with the persistent bus instance name (typically, `main` or `external` as specified <<persistent_bus,above>>)   |Size of the inflight queue (only valid in STICKY_EVENTS mode)   |30000   
+|Kill Bill Web Application   |org.killbill.server.multitenant   |Whether multi-tenancy is enabled   |true   
+|Kill Bill Web Application   |org.killbill.server.test.mode   |Whether to start in test mode   |false   
+|Kill Bill Web Application    |org.killbill.server.http.gzip   |Allow Kill Bill to return gzip json when Content-Encoding is set with gzip   |false   
+|Payment   |org.killbill.payment.plugin.timeout   |Timeout for each payment attempt   |30s   
+|===
+
+For a complete list of all the global properties, you can check the <<config_classes, config classes>> specified above.
+
+
+=== Java System Properties 
+
+It is also possible to specify the Kill Bill configuration properties as Java system properties. For example, the `org.killbill.dao.url` can be specified as a system property via `-Dorg.killbill.dao.url=<url>`.
```

Comment:
I don't believe this is still true anymore. Only selected System Properties can be specified now. We could search for `System.getProperty` calls in the codebase.

Reply:
okay, I inferred the above by taking a look at the  [DefaultKillbillConfigSource.loadPropertiesFromFileOrSystemProperties](https://github.com/killbill/killbill-platform/blob/780991187b7c90556340fcf00e9e1168a3db6d42/base/src/main/java/org/killbill/billing/platform/config/DefaultKillbillConfigSource.java#L119) which uses system properties in case a properties file is not specified. Is this code no longer used?

Reply:
Oh, you're right, it's still there! Good catch!

Back in the days, I believe we were _merging_ properties between System Properties and the configuration file. This is indeed not the case anymore, it's either file based or properties based.